boot/duneboot.ml

```diff
@@ -1163,15 +1163,15 @@ module Library = struct
           then (
             let extra_flags =
               if String.starts_with ~prefix:"blake3_" fn
-              then (
-                match architecture with
-                | "x86" | "i386" | "i486" | "i586" | "i686" ->
+              then
+                if Sys.cygwin || Sys.word_size == 32
```

Comment:
You should pass `word_size` to this function (like`ext_obj`) and obtain this value from the ocaml config.

Reply:
IINM the `Sys.os_type` values are determined at compile-time and not run-time, so I don't think they would affect cross-compilation.